CONTRIBUTING.md

```diff
@@ -0,0 +1,59 @@
+Package Contribution Guidelines
+===============================
+
+If you are submitting a new Spack package, please observe the
+following standards:
+
+1. Code must be [PEP-8 compliant](http://spack.readthedocs.io/en/latest/contribution_guide.html?highlight=pep-8#flake8-tests)
+
+2. At least one numeric (checksummable) version must be supplied.  If
+   this package does not yet have any official releases, create an
+   unofficial checksummable Spack-only release using something like:
+   ```
+   version('1.9.5.1.1',
+        git='https://github.com/jswhit/pyproj.git',
+        commit='0be612cc9f972e38b50a90c946a9b353e2ab140f')
+   ```
+
+2. All explicit versions in `depends_on()` statements must be for
+   checksummable versions.  If you need to depend on unreleased
+   versions of other packages, add unofficial releases to them too.
+
+3. If the package builds binaries... once you've built the package on
```

Comment:
I see; I agree, automation isn't completely out of the question.  But we would need to solve a bunch of problems:

1. The automated `ldd` would have to be done on a standardized test machine.  Real users (at least myself) specify the use of system-supplied libraries for SSL, MPI, compilers, X11, Qt, etc.  The presence of these libraries should therefore not be surprising.

2. If it's done in an automated way, how will that fit in with Travis?

3. When we turn this on, we will find a LOT of surprises, with MANY packages receiving new declared dependencies.  The problem is, if I run `ldd` and find a bunch of system dependencies, many of them could be caused by other Spack packages I depend on.  Until the whole Spack repo is "dependency-clean," there will be a lot of spurious system-supplied libraries I need to ignore.

For all these reasons, I think we're a ways from automated detection of undeclared dependencies.  In the meantime, I think package authors should run `ldd` and look over the results; or at least post them with the PR for others to look over.